docs/pbc/all/merchant-management/202407.0/marketplace/install-and-upgrade/tutorials-and-howtos/create-merchant-commission-calculator-type-plugin.md

```diff
@@ -0,0 +1,338 @@
+     *
+     * @return int
+     */
+    public function calculateSumPriceToPayPercentageMerchantCommissionAmount(
+        MerchantCommissionTransfer $merchantCommissionTransfer,
+        MerchantCommissionCalculationRequestItemTransfer $merchantCommissionCalculationRequestItemTransfer,
+        MerchantCommissionCalculationRequestTransfer $merchantCommissionCalculationRequestTransfer
+    ): int;
+}
+```
+
+**src/Pyz/Zed/MerchantCommission/Business/MerchantCommissionFacade.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\MerchantCommission\Business;
+
+use Generated\Shared\Transfer\MerchantCommissionCalculationRequestItemTransfer;
+use Generated\Shared\Transfer\MerchantCommissionCalculationRequestTransfer;
+use Generated\Shared\Transfer\MerchantCommissionTransfer;
+use Spryker\Zed\MerchantCommission\Business\MerchantCommissionFacade as SprykerMerchantCommissionFacade;
+
+/**
+ * @method \Spryker\Zed\MerchantCommission\Persistence\MerchantCommissionEntityManagerInterface getEntityManager()
+ * @method \Spryker\Zed\MerchantCommission\Persistence\MerchantCommissionRepositoryInterface getRepository()
+ * @method \Pyz\Zed\MerchantCommission\Business\MerchantCommissionBusinessFactory getFactory()
+ */
+class MerchantCommissionFacade extends SprykerMerchantCommissionFacade implements MerchantCommissionFacadeInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\MerchantCommissionTransfer $merchantCommissionTransfer
+     * @param \Generated\Shared\Transfer\MerchantCommissionCalculationRequestItemTransfer $merchantCommissionCalculationRequestItemTransfer
+     * @param \Generated\Shared\Transfer\MerchantCommissionCalculationRequestTransfer $merchantCommissionCalculationRequestTransfer
+     *
+     * @return int
+     */
+    public function calculateSumPriceToPayPercentageMerchantCommissionAmount(
+        MerchantCommissionTransfer $merchantCommissionTransfer,
+        MerchantCommissionCalculationRequestItemTransfer $merchantCommissionCalculationRequestItemTransfer,
+        MerchantCommissionCalculationRequestTransfer $merchantCommissionCalculationRequestTransfer
+    ): int {
+        return $this->getFactory()
+            ->createSumPriceToPayPercentageMerchantCommissionCalculatorType()
+            ->calculateMerchantCommissionAmount(
+                $merchantCommissionTransfer,
+                $merchantCommissionCalculationRequestItemTransfer,
+                $merchantCommissionCalculationRequestTransfer,
+            );
+    }
+}
+```
+
+## 4) Implement the calculator plugin
+
+This plugin should implement the `MerchantCommissionCalculatorPluginInterface` interface and provide the logic to calculate the merchant commission amount and correct persisting and reading the merchant commission amount.
+
+1. Method `getCalculatorType()` should return the type of the calculator as it will be stored in `spy_merchant_commission.calculator_plugin_type`.
```

Comment:
```suggestion
1. Method `getCalculatorType()` should return the type of the calculator as it will be stored in `spy_merchant_commission.calculator_plugin_type` database column.
```